snapcraft/services/provider.py

```diff
@@ -0,0 +1,32 @@
+# -*- Mode:Python; indent-tabs-mode:nil; tab-width:4 -*-
+#
+# Copyright 2024 Canonical Ltd.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License version 3 as
+# published by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+"""Snapcraft Lifecycle Service."""
+
+import os
+
+from craft_application import ProviderService
+from overrides import overrides
+
+
+class Provider(ProviderService):
+    """Snapcraft specialization of the Lifecycle Service."""
+
+    @overrides
+    def setup(self) -> None:
+        if build_info := os.getenv("SNAPCRAFT_BUILD_INFO"):
```

Comment:
this needs a test too

Reply:
Added a provider service unit test. Functionality is covered by the metadata spread tests.